lib/views/go_to_page.dart

```diff
@@ -69,20 +67,41 @@ class _GoToPageState extends State<GoToPage> {
                         decoration: InputDecoration(
                           contentPadding:
                               EdgeInsets.fromLTRB(15.0, 12.0, 20.0, 15.0),
-                          hintText: "origin...",
+                          hintText: "Choose starting location",
+                          hintStyle: TextStyle(color: Colors.grey),
                           border: OutlineInputBorder(
-                            borderRadius: BorderRadius.circular(10.0),
+                            borderRadius: BorderRadius.circular(
+                              10.0,
+                            ),
                           ),
                         ),
                         controller: this._originTypeAheadController,
                       ),
                       suggestionsCallback: (pattern) async {
-                        var roomTitles = new List<RoomCoordinate>();
-                        //TODO: add room list here
-                        for (var room in roomTitles) {
-                          //TODO: fix this for loop
+                        var roomTitles = new List<String>();
+                        for (var room in roomsList) {
+                          if (room.roomId
+                              .toUpperCase()
+                              .startsWith(pattern.toString().toUpperCase())) {
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                            roomTitles.add(room.roomId);
+                          }
                         }
-                        return roomTitles; //await BackendService.getSuggestions(pattern);
+                        return roomTitles;
```

Comment:
Can you explain these additions to me quickly? Why so much repetition?

Reply:
I believe this is an error as there should only be one addition

Reply:
that was for testing rendering failures but I got rid of them before committing. I don't see them on my branch which is weird, will be committed by Divesh to resolve it.

Reply:
The issue has been fixed